File: 0905 sort-array-by-parity/solution.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        // int[] arr=new int[nums.length];
        // int l=0,r=nums.length-1;
        // for(int i=0;i<nums.length;i++){
        //     if((nums[i] & 1)==0) arr[l++]=nums[i];
        //     else arr[r--]=nums[i];
        // }
        // return arr;






    int k=0;
    for(int i=0;i<nums.length;i++){
        if((nums[i] & 1)==0) {
            int temp=nums[k]; 
            nums[k++]=nums[i];
            nums[i]=temp;

        }
    }

return nums;

    }
}

File: 0905 sort-array-by-parity/solution.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int l = 0;
        for(int i=0;i<nums.size();i++) {
            if((nums[i] & 1) == 0) {
                int temp = nums[i];
                nums[i] = nums[l];
                nums[l++] = temp;
            }
        }
        return nums;
    }
};
